src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -1694,6 +1694,45 @@ MethodTableBuilder::BuildMethodTableThrowing(
         &pByValueClassCache, bmtMFDescs, bmtFP,
         &totalDeclaredFieldSize);
 
+    if (IsValueClass())
+    {
+        const void* pVal;                  // The custom value.
+        ULONG       cbVal;                 // Size of the custom value.
+        HRESULT hr = GetCustomAttribute(bmtInternal->pType->GetTypeDefToken(),
+            WellKnownAttribute::InlineArrayAttribute,
+            &pVal, &cbVal);
+
+        if (hr != S_FALSE)
+        {
+            if (bmtEnumFields->dwNumInstanceFields != 1)
+            {
+                // TODO: diagnostics, must have one field.
+                BuildMethodTableThrowException(IDS_CLASSLOAD_GENERAL);
+            }
+
+            if (cbVal >= (sizeof(INT32) + 2))
+            {
+                INT32 repeat = GET_UNALIGNED_VAL32((byte*)pVal + 2);
+                if (repeat > 0)
+                {
+                    bmtFP->NumInlineArrayElements = repeat;
+                    GetHalfBakedClass()->SetInlineArrayFlag();
+                }
+                else
+                {
+                    // TODO: diagnostics, repeat must be > 0
```

Comment:
Do we want a special diagnostic messages for this or `IDS_CLASSLOAD_GENERAL` is good enough?

Reply:
The bar for type loader error messages has been: 
- Good error message if it is something you can hit by writing C#. 
- Generic error message is ok if it is something that can be only hit by writing IL.

Reply:
oh, made a similar comment above. If there is some likelihood where customers might hit such cases a specific message would be helpful.

Reply:
> > Good error message if it is something you can hit by writing C#.
> 
> Yes, these can be hit, so need specific messages.

Isn't C# going to have syntax for this? If so, it's likely they're going to prevent people placing this attribute on random stuff like they do for `IsByRefLikeAttribute`. Would want to double check, but it's not an unreasonable request to make.

Reply:
I think for now we should assume that the attribute is valid for direct use. It is not as dangerous as `IsByRefLike` and is relatively easy to use correctly. There could be some use scenarios that are not covered by C# features.

If eventually we see that it is better to disallow, we can do that and make errors less friendly. 
I think for the time being, while C# features are in design/prototyping/development, nicer errors would be a good thing.